pkg/provisioner/controller.go

```diff
@@ -638,3 +648,62 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func updateSupported(old, new *v1alpha1.ObjectBucketClaim) bool {
+	// The only field supported for update is obc.spec.additionalConfig
+	if reflect.DeepEqual(new.Spec, old.Spec) {
+		return false
+	}
+	// create copy of old spec, and set the new spec's additionalConfig on it
+	oldspec := old.Spec.DeepCopy()
+	overwriteAdditionalConfig(new.Spec.AdditionalConfig, oldspec.AdditionalConfig)
+	if !reflect.DeepEqual(*oldspec, new.Spec) {
+		// new OBC spec has changed something other than additionalConfig
+		log.Error(nil, "invalid changes to OBC. only additionalConfig can be updated")
+		return false
+	}
+	return true
+}
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+
+	log.Info("Additional config changed, hence updating OB")
+	tmp := make(map[string]string)
+
+	tmp = ob.Spec.Endpoint.AdditionalConfigData
+	overwriteAdditionalConfig(obc.Spec.AdditionalConfig, ob.Spec.Endpoint.AdditionalConfigData)
+	err = c.provisioner.Update(ob)
+	if err != nil {
+		log.Error(err, "updating from Provisioner failed")
+		return err
+	}
+	err = wait.PollImmediate(defaultRetryTimeout, defaultRetryBaseInterval, func() (done bool, err error) {
+		_, err = updateObjectBucket(ob, c.libClientset)
+		if err != nil {
+			log.Error(err, "updating OB failed, retrying...")
+			return false, err
+		}
+		return true, nil
+	})
+
+	if err != nil {
+		log.Error(err, "updating OB failed, reverting provisioner to original value")
+		overwriteAdditionalConfig(tmp, ob.Spec.Endpoint.AdditionalConfigData)
+		err = c.provisioner.Update(ob)
+	}
```

Comment:
I think it's best to keep the `if err != nil` connected visually (no space between). I would separate the provisioner update from the OB update with a newline but not the error checking on updating the OB

```suggestion

	err = wait.PollImmediate(defaultRetryTimeout, defaultRetryBaseInterval, func() (done bool, err error) {
		_, err = updateObjectBucket(ob, c.libClientset)
		if err != nil {
			log.Error(err, "updating OB failed, retrying...")
			return false, err
		}
		return true, nil
	})
	if err != nil {
		log.Error(err, "updating OB failed, reverting provisioner to original value")
		overwriteAdditionalConfig(tmp, ob.Spec.Endpoint.AdditionalConfigData)
		err = c.provisioner.Update(ob)
	}
```